### PR TITLE
📦 New Feature: Colors to black functionality

### PR DESCRIPTION
**Pull request by mikesalvia**
This PR allows for users to switch color in the collected image into black. This was very helpful in verifying "painted on" captions in ExoPlayer and AVPlayer for Android and iOS respectively.

⚠️ Probably best that this is pulled and taken for a test drive.

### CODE REVIEW

**Comment by mikesalvia** [comment]
@wswebcreation will look into the failure to see why this is happening and see about some unit tests

File: lib/helpers/afterScreenshot.interfaces.ts [hunk 1]
```diff
@@ -21,11 +21,13 @@ export interface AfterScreenshotOptions {
   // The file name options object
   fileName: ScreenshotFileNameOptions;
   // Elements that need to be hidden (visibility: hidden) before saving a screenshot
-  hideElements: (HTMLElement | HTMLElement[])[];
+  hideElements: HTMLElement[];
```
Comment (wswebcreation) [inline lib/helpers/afterScreenshot.interfaces.ts]: Why did you change the interface here? `hideElements` and `removeElements` will always be an array that hold or single elements, or a combination of single elements and an array of elements (like `[$('something'), $$('something-else')]`)

File: package.json [hunk 2]
```diff
@@ -32,7 +32,8 @@
   "dependencies": {
     "canvas": "^2.6.0",
     "chalk": "^2.4.2",
-    "fs-extra": "^7.0.1"
+    "fs-extra": "^7.0.1",
+    "gm": "^1.23.1"
```
Comment (wswebcreation) [inline package.json]: I try to prevent using _non standard_ environment modules like `imagemagick` and `graphicsmagick`. Not every user/customer has this standard installed in their CI pipeline which makes this break in pipelines.

ResembleJS also provides a `ignoreColors` option, doesn't that work?

Comment (mikesalvia) [reply package.json]: I will have to take a look and I will look at commments soon!

Comment (mikesalvia) [reply package.json]: OK so `ignoreColors` does not work for me, that basically will compare at grayscale.

The usage of this solves is reading "painted on" captions from a video frame where the video frame may have slight differences.

Our app is a testbed for our video player, in that player, there are 608 captions painted on, over video. In order to validate them, we need to extract captions. However, since there are small changes to the background image even ignoring colors doesn't help. What this does is take any color and floor it to black (not grays), leaving a pure black image with white text to compare.

Comment (mikesalvia) [reply package.json]: Also, these libraries are only looked for and used if this option is enabled... I understand non-standard libraries but if you use this option you will need to install the required components.

**Comment by mikesalvia** [comment]
Screen Shot 2019-11-04 at 1 44 09 PM
This is what the feature essentially accomplishes.

**Comment by mikesalvia** [comment]
@wswebcreation Wanted to see your thoughts on this feature. I have realized it will rely on GraphicsMagick if installed but fails gracefully if it is not installed.

**Comment by wswebcreation** [comment]
Hi @mikesalvia 

I've been thinking about this feature and I do believe this is a really great feature. Still I wonder if this is needs to be done with other libs than ResembleJS. Secondly this is a really unique case (IMHO, sorry) for which we add some extra dependencies (even though you made it fail gracefully if it is not installed).

By adding GraphicsMagick we open a complete new door in doing the comparison, which also increases the change in getting more questions from people on how to use this, if it supports more options and so on.

My proposal would be the following:

1. Or add this feature in ResembleJS (I'm using a fork) and keep the extra dependencies to 0
2. Open the door for using GraphicsMagick for comparison, but that means we need to have an extra maintainer for this module because I don't have the time to also support that. The extra maintainer needs to _take over_ all GraphicsMagick releated questions / issues in all modules that use this core and also make sure all docs are up to date, explain what you can do with it and so on

What do you think

**Comment by wswebcreation** [comment]
Going to close it due to inactivity
